### PR TITLE
fixed issue 1642 - bug in empty strings in compounds

### PR DESCRIPTION
**Pull request by kzangeli**
Fixes issue #1642

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -10,3 +10,4 @@
 - Fix: enforcing "http" or "https" schema and better detection of missing port at URLs parsing (Issue #1652)
 - Add: POST /v2/subscriptions operation (without expression evaluation yet) (Issue #1316)
 - Add: DELETE /v2/subscriptions operation (Issue #1654)
+- Fix: fixed a bug about empty strings in compounds (Issue #1642)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I'd suggest the following (more precise):

"Fix: fixed wrong interpretation of empty string keyvalues as empty objects in compounds (Issue #1642)".

File: test/functionalTest/cases/1308_subscription_cache/sub_expired_and_updated_then_cache_refresh.test [hunk 2]
```diff
@@ -289,9 +289,9 @@ Date: REGEX(.*)
 
 {
     "ids": "REGEX(.*)",
-    "inserts": REGEX((2|3)),
+    "inserts": REGEX((2|3|4)),
```
Comment (fgalan) [inline test/functionalTest/cases/1308_subscription_cache/sub_expired_and_updated_then_cache_refresh.test]: Not really related with this PR, right?

Comment (kzangeli) [inline test/functionalTest/cases/1308_subscription_cache/sub_expired_and_updated_then_cache_refresh.test]: Nothing to do with the PR, correct.
This functest failed sporadically.
Now it should not fail anymore ...

NTC

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in de077ea

**Comment by fgalan** [comment]
LGTM

**Pull request merged**